examples/js/postprocessing/DepthPass.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * @author jjxtra / https://www.digitalruby.com
+ */
+
+THREE.DepthPass = function(scene, camera)
+{
+	THREE.Pass.call(this);
+
+	this.scene = scene;
+	this.camera = camera;
+	this.needsSwap = false;
+};
+
+THREE.DepthPass.prototype = Object.assign( Object.create( THREE.Pass.prototype ), {
+
+	constructor: THREE.DepthPass,
+
+	render: function ( renderer, writeBuffer, readBuffer /*, deltaTime, maskActive */ ) {
+
+		this.scene.overrideMaterial = null;
+		renderer.clearDepth();
+		renderer.setRenderTarget(this.renderToScreen ? null : readBuffer);
+		renderer.context.colorMask(false, false, false, false);
```

Comment:
We generally avoid making gl calls directly because it can mess up the renderer's internal state. You may have to do this:
```js
renderer.state.buffers.color.setMask( ... );
```

Reply:
Updated

Reply:
It is sort of an awkward API, but it works for you, right?

Reply:
It is working, I will think about a demo, for now it is a contract job that I cannot post anything but the depth pass, which I am using for another personal side project, so happy to share.

Reply:
Would it work to add it to one of the boxes in [this example](https://threejs.org/examples/webgl_postprocessing_advanced.html)?

Reply:
Could do, the main use case I have found is post processing transparent objects like lines and particles that wrap around, through and behind solid objects.

Reply:
Something cool like that would be ... cool.  :-)